src/abi/anchor.rs

```diff
@@ -157,7 +157,14 @@ fn idl_instructions(
                 vec![IdlAccountItem::IdlAccount(IdlAccount {
                     name: "dataAccount".to_string(),
                     is_mut: true,
-                    is_signer: func.has_payer_annotation(),
+                    /// With @payer annotation, the account is created on-chain and needs a signer. The client
+                    /// provides an address which does not exist yet, so SystemProgram.CreateAccount is called
```

Comment:
```suggestion
                    /// provides an address that does not exist yet, so SystemProgram.CreateAccount is called
```
This is a restrictive clause.